File: lista/Python/questao13.py
"""
Adicione um método estático à classe Matematica que calcula o
fatorial de um número. Em Python, utilize @staticmethod.
"""
class Matematica:
    @staticmethod
    def fatorial(n):
        if n < 0:
            raise ValueError("O fatorial não é definido para números negativos.")
        elif n == 0 or n == 1:
            return 1
        else:
            resultado = 1
            for i in range(2, n + 1):
                resultado *= i
            return resultado

def main():

    numero = 5
    resultado = Matematica.fatorial(numero)
    print(f"O fatorial de {numero} é {resultado}")

if __name__ == "__main__":
    main()

File: lista/GO/questao13/questao13.go
package main

import (
	"fmt"
)

type Matematica struct{}

func (Matematica) Fatorial(n int) int {
	if n < 0 {
		return -1
	}
	if n == 0 || n == 1 {
		return 1
	}
	resultado := 1
	for i := 2; i <= n; i++ {
		resultado *= i
	}
	return resultado
}

func main() {
	m := Matematica{}

	numero := 5
	fatorial := m.Fatorial(numero)

	if fatorial != -1 {
		fmt.Printf("O fatorial de %d é: %d\n", numero, fatorial)
	} else {
		fmt.Println("Erro")
	}
}
